chainer/datasets/sub_dataset.py

```diff
@@ -0,0 +1,157 @@
+import numpy
+import six
+
+from chainer.dataset import dataset_mixin
+
+
+class SubDataset(dataset_mixin.DatasetMixin):
+
+    """Subset of a base dataset.
+
+    SubDataset defines a subset of a given base dataset. The subset is defined
+    as an interval of indexes, optionally with a given permutation.
+
+    If ``order`` is given, then the ``i``-th example of this dataset is the
+    ``order[start + i]``-th example of the base dataset, where ``i`` is a
+    non-negative integer. If ``order`` is not given, then the ``i``-th example
+    of this dataset is the ``start + i``-th example of the base dataset.
+    Negative indexing is also allowed: in this case, the term ``start + i`` is
+    replaced by ``finish + i``.
+
+    Args:
+        dataset: Base dataset.
+        start (int): The first index in the interval.
+        finish (int): The next-to-the-last index in the interval.
+        order (sequence of ints): Permutation of indexes in the base dataset.
+            If this is None, then the ascending order of indexes is used.
+
+    """
+    def __init__(self, dataset, start, finish, order=None):
+        if start < 0 or finish > len(dataset):
+            raise ValueError('subset overruns the base dataset.')
+        self._dataset = dataset
+        self._start = start
+        self._finish = finish
+        self._size = finish - start
+        self._order = order
+
+    def __len__(self):
+        return self._size
+
+    def get_example(self, i):
+        if i >= 0:
+            if i >= self._size:
+                raise IndexError('dataset index out of range')
+            index = self._start + i
+        else:
+            if i < -self._size:
+                raise IndexError('dataset index out of range')
+            index = self._finish + i
+
+        if self._order is not None:
+            index = self._order[index]
+        return self._dataset[index]
+
+
+def split_dataset(dataset, split_at, order=None):
+    """Splits a dataset into two subsets.
+
+    This function creates two instances of :class:`SubDataset`. These instances
+    do not share any examples, and they together cover all examples of the
+    original dataset.
+
+    Args:
+        dataset: Dataset to split.
```

Comment:
I think "to split" is valid (like "things to do"), but I'm not so confident... do we have to consult a native speaker?